gitlab4j-api/src/main/java/org/gitlab4j/api/JobApi.java

```diff
@@ -292,18 +292,38 @@ public Stream<Job> getJobsStream(Object projectIdOrPath, long pipelineId, Boolea
     }
 
     /**
-     * Retrieve the job corresponding to the <code>$CI_JOB_TOKEN</code> environment variable (Using a {@link org.gitlab4j.models.Constants.TokenType#JOB_TOKEN} authentication).
+     * Retrieve the job corresponding to the <code>$CI_JOB_TOKEN</code> environment variable (Using a {@link TokenType#JOB_TOKEN} authentication).
      *
      * <pre><code>GitLab Endpoint: GET /job</code></pre>
      *
-     * @return a single job
+     * @return a single job corresponding to the token used for the authentication
      * @throws GitLabApiException if any exception occurs during execution
      */
     public Job getJob() throws GitLabApiException {
+        TokenType tokenType = getApiClient().getTokenType();
+        if (tokenType != TokenType.JOB_TOKEN) {
+            throw new IllegalStateException(
+                    "This method can only be called with a " + TokenType.JOB_TOKEN + " authentication");
+        }
         Response response = get(Response.Status.OK, null, "job");
         return (response.readEntity(Job.class));
     }
 
+    /**
+     * Retrieve the job corresponding to the <code>$CI_JOB_TOKEN</code> environment variable.
```

Comment:
This works only without any authentication, maybe this information would be needed in the javadoc